lib/pbench/server/filetree.py

```diff
@@ -259,21 +258,6 @@ def create(cls, tarball: Path, controller: "Controller") -> "Tarball":
             # log it but do not abort
             controller.logger.error("Unable to set SELINUX context for {}: {}", name, e)
 
-        # To get the new tarball into the server pipeline, we start with a
-        # symlink in the TODO state directory.
-        #
-        # NOTE: there's not much we can do if the link fails, except report it.
-        # The solution is for someone to manually create the link to oil the
-        # tool chain machinery. It doesn't make sense to fail and back out at
-        # this point, and there's no place to report the error aside from the
-        # log.
-        try:
-            controller.link(destination, "TODO")
```

Comment:
Sorry, but I'm still stuck on this.  😕 

If we have an existing set of already-uploaded tarballs which we want to cause to be indexed, how do we do that?  (And, for the sake of argument, lets assume that the database doesn't know about them (so it's not a "re-index" case)...e.g., suppose that we've suffered a catastrophic hardware failure or we're creating a totally new server deployment, and we've restored all the tarballs (but not the databases) from backup, and now we need to re-ingest them.)  Since there is no `PUT` operation, and since this change removes the symlink-based tickle for the unpacker, do we have no story for that case?

(I'm not saying that there has to be a story -- in the unlikely event that this happens, I'm sure we could put something together to bootstrap the Server -- but, I just want confirmation that we're removing something here that we don't have a replacement for.  Thanks!)

Reply:
The new `pbench-state-manager` allows manipulating `did`/`enabled` manually, so if we really want to play with the idea of faking dataset import we certainly could. However even if we have local datasets it'd be a lot cleaner to use `curl` to `PUT` them properly.

Regardless, especially as "file tree" becomes "cache manager", dealing with this sort of control flow issue just doesn't belong here.